src/components/sidenav/sidenav.ts

```diff
@@ -0,0 +1,353 @@
+  }
+  @HostBinding('class.md-sidenav-opened') private get isOpened_() {
+    return this.opened_ && !this.transition_;
+  }
+  @HostBinding('class.md-sidenav-end') private get isEnd_() {
+    return this.align == 'end';
+  }
+  @HostBinding('class.md-sidenav-side') private get modeSide_() {
+    return this.mode == 'side';
+  }
+  @HostBinding('class.md-sidenav-over') private get modeOver_() {
+    return this.mode == 'over';
+  }
+  @HostBinding('class.md-sidenav-push') private get modePush_() {
+    return this.mode == 'push';
+  }
+
+  /**
+   * This is public because we need it from MdSidenavLayout, but it's undocumented and should
+   * not be used outside.
+   * @returns {number}
+   * @private
+   */
+  public get width_() {
+    if (this.elementRef_.nativeElement) {
+      return this.elementRef_.nativeElement.offsetWidth;
+    }
+    return 0;
+  }
+
+  private transition_: boolean = false;
+  private openPromise_: Promise<void>;
+  private openPromiseResolve_: () => void;
+  private openPromiseReject_: () => void;
+  private closePromise_: Promise<void>;
+  private closePromiseResolve_: () => void;
+  private closePromiseReject_: () => void;
+}
+
+
+
+/**
+ * <md-sidenav-layout> component.
+ */
+@Component({
+  selector: 'md-sidenav-layout',
+  directives: [MdSidenav],
+  templateUrl: './components/sidenav/sidenav.html',
+  styleUrls: ['./components/sidenav/sidenav.css'],
+})
+export class MdSidenavLayout implements AfterContentInit {
+  @ContentChildren(MdSidenav) private drawers_: QueryList<MdSidenav>;
+
+  get start() { return this.start_; }
+  get end() { return this.end_; }
+
+  constructor(@Optional() @Host() private dir_: Dir) {
+    if (dir_) {
+      dir_.dirChange.subscribe(() => this.validateDrawers_());
+    }
+  }
+
+  ngAfterContentInit() {
+    // On changes, assert on consistency.
+    this.drawers_.changes.subscribe(() => this.validateDrawers_());
+    this.validateDrawers_();
+  }
+
+
+
+  private start_: MdSidenav;
+  private end_: MdSidenav;
+  private right_: MdSidenav;
+  private left_: MdSidenav;
+
+  private validateDrawers_() {
+    this.start_ = this.end_ = null;
+    if (this.drawers_.length === 0) {
+      throw new MdMissingSidenavException();
+    }
+
+    this.drawers_.toArray().forEach(drawer => {
+      if (drawer.align == 'end') {
+        if (this.end_) {
+          throw new MdDuplicatedSidenavException('end');
+        }
+        this.end_ = drawer;
+      } else {
+        if (this.start_) {
+          throw new MdDuplicatedSidenavException('start');
+        }
+        this.start_ = drawer;
+      }
+    });
+
+    this.right_ = this.left_ = null;
+    this.left_ = this.start_;
+    this.right_ = this.end_;
+
+    // Detect if we're LTR or RTL.
+    if (!this.dir_ || this.dir_.dir == 'ltr') {
+      this.left_ = this.start_;
+      this.right_ = this.end_;
+    } else {
+      this.left_ = this.end_;
+      this.right_ = this.start_;
+    }
+  }
+
+  private onBackdropClick_() {
```

Comment:
Rather than naming the function based on how it's going to be called, can we name it based on what it does? In this case, something like `closeModalSidenavs` ?